database/fingertips-db/StoredProcedures/GetDescendantAreas.sql

```
@@ -0,0 +1,99 @@
+                FROM 
+                    Areas.Areas area
+                WHERE 
+                    area.AreaCode = @RequestedAncestorAreaCode
+                AND 
+                    area.AreaTypeKey <> @RequestedAreaType
+            ),
+            TargetAreaType AS (
+                SELECT 
+                    *
+                FROM 
+                    Areas.AreaTypes AT
+                WHERE 
+                    --- this is the taget areaType
+                    at.AreaTypeKey = @RequestedAreaType
+            ),
+        recursive_cte(
+            AreaKey,
+            AreaCode,
+            AreaName,
+            AreaTypeKey,
+            AreaLevel,
+            ParentAreaKey
+        ) AS (
+            SELECT 
+                ar.ChildAreaKey AS AreaKey,
+                a.AreaCode,
+                a.AreaName,
+                a.AreaTypeKey,
+                at2.[LEVEL] AS AreaLevel,
+                ar.ParentAreaKey
+            FROM 
+                Areas.AreaRelationships ar
+            JOIN 
+                Areas.Areas a 
+            ON 
+                a.AreaKey = ar.ChildAreaKey
+            JOIN 
+                StartingArea sa 
+            ON 
+                ar.ParentAreaKey = sa.AreaKey
+            JOIN 
+                Areas.AreaTypes at2 
+            ON 
+                at2.AreaTypeKey = a.AreaTypeKey
+            UNION ALL
+            SELECT 
+                ar.ChildAreaKey AS AreaKey,
+                a.AreaCode,
+                a.AreaName,
+                a.AreaTypeKey,
+                at2.[LEVEL] AS AreaLevel,
+                ar.ParentAreaKey
+            FROM 
+                Areas.AreaRelationships ar
+            JOIN 
+                Areas.Areas a 
+            ON 
+                a.AreaKey = ar.ChildAreaKey
+            JOIN 
+                Areas.AreaTypes at2 
+            ON 
+                at2.AreaTypeKey = a.AreaTypeKey
+            INNER JOIN 
+                recursive_cte 
```

Is this safe?  never use recursive because , however it would be nice to have a meaningful name for readibility.

---

This was rolled back because it's important if the recursion exceeds limits that the SQL should fail rather than simply return.